distributed/worker_client.py

```diff
@@ -53,22 +53,26 @@ def worker_client(timeout=None, separate_thread=True):
 
     worker = get_worker()
     client = get_client(timeout=timeout)
-    if separate_thread:
-        duration = time() - thread_state.start_time
-        secede()  # have this thread secede from the thread pool
-        worker.loop.add_callback(
-            worker.handle_stimulus,
-            SecedeEvent(
-                key=thread_state.key,
-                compute_duration=duration,
-                stimulus_id=f"worker-client-secede-{time()}",
-            ),
-        )
-
-    yield client
-
-    if separate_thread:
-        rejoin()
```

Comment:
previously this didn't secede if there was an exception in `with worker_client()`